gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeUtils.java

```diff
@@ -186,4 +190,129 @@ private static boolean containsNotOrOrFilter(Filter filter) {
 
     return false;
   }
+
+  /**
+   * Computes common set of sources, if both filters and order bys are requested on the same source
+   * sets
+   *
+   * Look at {@link ExecutionTreeUtils#getIntersectingSourceSets(java.util.Map, java.util.Map)}
+   */
+  public static Set<String> getSourceSetsIfFilterAndOrderByAreFromSameSourceSets(
+      ExecutionContext executionContext) {
+
+    Map<String, Set<String>> sourceToFilterAttributeMap = executionContext.getSourceToFilterAttributeMap();
+    Map<String, Set<String>> sourceToSelectionOrderByAttributeMap = executionContext.getSourceToSelectionOrderByAttributeMap();
+    Map<String, Set<String>> sourceToMetricOrderByAttributeMap = executionContext.getSourceToMetricOrderByAttributeMap();
+
+    // merges sourceToSelectionOrderByAttributeMap and sourceToMetricOrderByAttributeMap
+    Map<String, Set<String>> sourceToOrderByAttributeMap =
+        Stream.concat(
+            sourceToSelectionOrderByAttributeMap.entrySet().stream(),
+            sourceToMetricOrderByAttributeMap.entrySet().stream())
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    Map.Entry::getValue,
+                    (v1, v2) -> {
```

Comment:
Can we make sure that all scenarios in this class are unit tested?

Reply:
yep. I have added unit tests for all the scenarios